### ISSUE TITLE
Prep 0.14.0 Release

### ISSUE THREAD

**Issue by j5awry** [noirbizarre/flask-restplus#743]
We've accumulated a fair set of fixes. There are 2 more PRs in the pipeline that I think should get grabbed in

#739 
#741 

After those, and any other intermediary PRs, we should cut a release. Some  prep things we should do:

1. ensure the changelog has all the changes in it.
2. do a final check that all tests are passing, and our own projects are happily running on the new version.

**Comment by SteadBytes** [comment]
Agreed, thanks @j5awry . I was planning on pinging again about this on https://github.com/noirbizarre/flask-restplus/issues/721 . I'll have a check through and see if there's any other PRs worth adding in for this release 👍

**Comment by SteadBytes** [comment]
In addition to those you mentioned it would be good to get https://github.com/noirbizarre/flask-restplus/pull/708 in . It's been open a while and the original feature requester approved of the implementation.

Other than that most other PRs seem to be incomplete or not fully reviewed/required

**Comment by j5awry** [comment]
@ziirish merged #708, so it's good now. Sorry I had to quiet down restplus emails a bit the last month or so (had to pull back a bit to focus on other stuff). Also, yes to #721. It can go in the same PR that does changelog and uprev IMO.

**Comment by j5awry** [comment]
I think we're good for a release on this commit:

d4ace9cabbbe63b49907a5dfcf46597b467470ee

I don't see docs on doing a release. I see bumpr.rc, which indicates that bumpr has been used. Sounds like it's still a task for @noirbizarre 

Axel, could you do a release off the commit above?

**Comment by SteadBytes** [comment]
We just need to push a new tag and the release is automated from that -
@noirbizarre changed it a while ago iirc 😁

**Comment by j5awry** [comment]
Please add all that info to the CONTRIBUTION.md or other relevant doc. Let's not leave that to institutional knowledge. I saw in a ticket asking for automating release, but never saw the final outcome.

**Comment by SteadBytes** [comment]
Yes I agree, that should have been when it was implemented. @noirbizarre
made the change so I'm not 100% sure on the details but I'll see what I can
find and get a PR up for it 👍

**Comment by SteadBytes** [comment]
Hmmm, so I've found the message on Gitter where we discussed this with @noirbizarre:
> The release is already automated using bumpr. I will just switch the process from "release from my computer and push tag" to "push tag and release from travis".

Shortly after this the 0.13.0 release was made by @noirbizarre , however I cannot see any changes to `travis.yml` to support this. I have a feeling that @noirbizarre made the release manually and didn't implement the "push tag and release from travis" :thinking:

**Comment by ziirish** [comment]
I think we could merge #637 as well.

**Comment by jonathanronen** [comment]
I'll be looking forward to this release  including #673! :+1:

**Comment by SteadBytes** [comment]
To keep this issue up to date with current state of affairs, as discussed on [Gitter](https://gitter.im/noirbizarre/flask-restplus) I reached out to @noirbizarre via Twitter DMs on 10/11/2019 to try and get his assistance with making a release. On 21/11/2019 I received a reply stating that he will review the release issue (this one), make the release and then automate subsequent releases so the maintainers can manage releases without his direct intervention. We have received no further communication and (clearly) the release has not been made.

**Comment by rogeriolino** [comment]
Any idea about the release date?

**Comment by CrazyBonze** [comment]
@SteadBytes any way we can get someone else with @noirbizarre privileges for the project so we dont have the to worry about the bus factor https://en.wikipedia.org/wiki/Bus_factor

**Comment by SteadBytes** [comment]
@CrazyBonze That's what we're trying to do, however we need @noirbizarre to give those privileges in the first place :thinking:

**Comment by SteadBytes** [comment]
@rogeriolino Unfortunately no, sorry. The code is ready to go, however the other maintainers and I need @noirbizarre to make the actual release and (as previously stated) he has yet to make the release and/or communicate back with us why not. Apologies again, we'll keep this thread updated with progress if any is made :+1:

**Comment by cal97g** [comment]
@SteadBytes Why don't we make a fork for now with the aim of being an up to date drop in replacement for flask-restplus?

This has the advantage of a) releasing updates to the community while b) possibly giving the owner some encouragement to maintain this project or hand it to somebody who will.

**Comment by j5awry** [comment]
+1 for fork. It's a bsd-3 license, so we can fork and even create a new thing in Pypi, as long as we keep all licensing in place, and state it's a derivative work. Otherwise, we can consider this dead, and start looking to move to other frameworks (I've mentioned where I'm pointing folks in other places)

I'll note that if we're thinking of forking, it might be worth planning the project again on "what a 1.0" would look like.

**Comment by SteadBytes** [comment]
@cal97g Yep, I think that's a good suggestion 👍

@j5awry , as long as we're good on the licence side the only issue is what to call it - I don't think I can bring myself to make a Flask-RestPlusPlus 😝

Also, I agree, some planning and design for a 1.0 is definitely a good idea, if this is feasible we need some goals to work to other than "fix bugs and add new bits here and there" 😁

**Comment by cal97g** [comment]
Presumably we can take inspiration from the issue still present here both for bug fixes and 1.0.0 features. @SteadBytes is better placed than me to recommend these.

I've started a new project and invited steadybytes. I've cloned this repo, keeping history intact.

I did go with Flask-RestPlusPlus but we can rename it later, or keep it  🤣

**Comment by SteadBytes** [comment]
Thanks for the invite @cal97g  😊 However I'm going to hold off until some of the other maintainers have had their say, we've all invested a lot of time in this and I don't want to start a fork without their input. @j5awry has mentioned on Gitter that he needs a little bit of time to evaluate his available commitment for the future (real life is more important!) and we're yet to hear from the others. I will wait until this weekend and then move forwards - does that sound ok?

**Comment by cal97g** [comment]
@SteadBytes yep sounds good to me. I'm just keen to make things happen as I'm now using this in production.

**Comment by SteadBytes** [comment]
@cal97g Still waiting to hear back from @j5awry (no rush), @ziirish is also in favour of forking and I haven't heard from the others. 

I'm aware that you've forked the project yourself and invited me (thanks again btw), however I'm considering whether it might be best for the existing maintainers to start the new fork 🤔 I am by no means suggesting that you're not a valid and welcomed contributor, however it sounds like you're reasonably new to the project and i think it makes sense for the existing maintainers to start moving it forward with the new fork. Again, your (and others) contributions will be of course welcome!

**Comment by cal97g** [comment]
@SteadBytes I don't disagree, once we have some sort of consensus here I will be happy to delete / handover the fork and move with whatever the current maintainers propose.

**Comment by ziirish** [comment]
Hi,

As said on gitter, we should have heard from @noirbizarre since we tried to get in touch for weeks/months.
I'm still waiting for news until the end of the year.
After that, if we still didn't head from him, then let's start a fork.

We'll need an org to make things more sustainable. If you have any idea for a name feel free to propose it.
We'll also need a new module name... I'm proposing "Flask-RestSwag"

What do you think?

**Comment by SteadBytes** [comment]
Thanks @cal97g that's very understanding and reasonable of you! 

@ziirish yes I agree an org might be a good idea 👍 With regards to the name I don't have any great ideas right now, but I'm not sure about Flask-RestSwag either 🤣

**Comment by SVilgelm** [comment]
Hi Guys,
What have you decide? From my experience, it's better to create an organization. Since there is a request to implement OpenAPI 3, I think the name of the project and organization could be `Flask-OpenAPI`. It's better that `RestSwag` :)

**Comment by SteadBytes** [comment]
Hey @SVilgelm , some discussion is happening over on Gitter whilst waiting for other maintainers to confirm their availability/willingness to go ahead now that the new year is here. 

I agree 100% that an organisation is the way to go 👍 I'm thinking a `restplus` (or whatever the new name is ) organisation which gives scope to producing other `framework-restplus` libraries.

**Comment by SVilgelm** [comment]
@SteadBytes thanks, i did read the conversation on gitter and as i said. Would be glad to do this for you, if  you have no time for such work -  forking, creating pypi, configuring all needed actions to check  pr and release the versions and etc...

**Comment by ziirish** [comment]
As stated earlier. Since the grace period is over, let's go ahead and fork this project.

To be 100% transparent, I'm a newbie with all the OpenAPI/Swagger things so I'm offering to focus my contributions on the core, the tests and all the release tooling if that can be of any help.
